test/integration/controller/jobs/job/job_controller_test.go

```diff
@@ -1145,7 +1145,7 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", ginkgo.Orde
 		highPriorityClass := testing.MakePriorityClass("high").PriorityValue(100).Obj()
 		gomega.Expect(k8sClient.Create(ctx, highPriorityClass))
 		ginkgo.DeferCleanup(func() {
-			gomega.Expect(k8sClient.Delete(ctx, highPriorityClass)).To(gomega.Succeed())
+			gomega.Eventually(func() error { return k8sClient.Delete(ctx, highPriorityClass) }, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
Why do we need this?

Reply:
I saw a flaky behavior on this on during my local test that was caused by client throttling and  there  is no legitimate expectation from our side to be able to delete that priority class on the first try.